### PR TITLE
Feat: Account colour customization

### PR DESCRIPTION
**Pull request by jeeanribeiro**
# Description of change

Adds accounts colors customization

Closes https://github.com/iotaledger/firefly/issues/2052

## Type of change

- Enhancement (a non-breaking change that adds functionality)

## How the change has been tested

- Create wallet
- Edit existing account and change color and pattern
- Create a new account and choose color and pattern
- Close the app and login again
- Hover AccountTiles (the chosen color and pattern should show)
- Edit account (color and patterns saved should be selected)
- Use mouse hover and click or keyboard tab and enter to select colors and patterns options

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/shared/locales/pt-BR.json [hunk 1]
```diff
@@ -528,6 +528,10 @@
                     "description": "A rede de incentivos trará grandes inovações para IOTA. O que quer que aconteça, acontecerá. Como o Shimmer evoluirá depende de você."
                 }
             }
+        },
```
Comment (begonaalvarezd) [inline packages/shared/locales/pt-BR.json:531]: we should undo all the changes in this file, the CI and the community in crowdin will take care of the translation 🙏🏼

**Review by nicole-obrien — CHANGES_REQUESTED** [review]
Currently if you are customising an existing wallet, you can only click save if the name has changed, even if you change colour or pattern.

Wallet switcher colours seem to break. Although this will be changed in the future, it is worth quickly fixing this too.

**Review by nicole-obrien — CHANGES_REQUESTED** [review]
There are still some issues:

1. The graphs are no longer using the custom colour in this branch, here is what it looks like in 1.3.3 followed by this branch.

Screenshot 2022-01-27 at 12 06 03
Screenshot 2022-01-27 at 12 08 28

2. The colours are still not random when a profile is restored through backup, mnemonic or when accounts are found in balance finder. This is present in 1.3.3.

File: packages/shared/components/AccountTile.svelte [hunk 2]
```diff
@@ -27,21 +27,26 @@
...
     export let size = 'm' // m, l
     export let hidden = false
     export let disabled = false
+    export let classes = ''
     export let onClick = (): void | string => ''
+    export let disabledHover = false
 
     $: darkModeEnabled = $appSettings.darkMode
 
+    $: textColor = isBright(color) ? 'gray-800' : 'white'
```
Comment (nicole-obrien) [inline packages/shared/components/AccountTile.svelte:48]: `isBright(color) ? 'gray-800' : 'white'`
This piece of code is used in a lot of files so potentially could be abstracted out.

File: packages/shared/components/ColorPicker.svelte [hunk 3]
```diff
@@ -0,0 +1,101 @@
+<script lang="typescript">
```
Comment (nicole-obrien) [inline packages/shared/components/ColorPicker.svelte:1]: Super confusing file, with the various states the styles can be in. Would be good to clean this up in the future and maybe separate the custom picker as its own component.

File: packages/shared/lib/helpers.ts [hunk 4]
```diff
@@ -251,3 +251,21 @@ export function deepCopy(obj: unknown): unknown {
         }, {})
```
Comment (nicole-obrien) [inline packages/shared/lib/helpers.ts:251]: We can do this in the major refactor coming, but we should have separate helper files for different topics under a helper folder, i.e. this is related to colours.

File: packages/shared/lib/profile.ts [hunk 5]
```diff
@@ -12,8 +18,13 @@ import { AvailableExchangeRates } from './typings/currency'
 import type { WalletAccount } from './typings/wallet'
 import { getOfficialNetworkConfig } from './network'
 import { NetworkConfig, NetworkType } from './typings/network'
+import { account } from './typings'
 
 export const activeProfileId = persistent<string | null>('activeProfileId', null)
+export interface ProfileAccount {
```
Comment (nicole-obrien) [inline packages/shared/lib/profile.ts:24]: Should separate interfaces into type files. Will be covered by the big refactor.

**Review by nicole-obrien — APPROVED** [review]
Approved with some suggestions for the future. 😄

**Pull request merged**